src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1336,8 +1343,11 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
         ucs_debug("got error on DC flow-control endpoint, iface %p: %s", iface,
                   ucs_status_string(ep_status));
     } else {
-        uct_dc_mlx5_iface_set_ep_failed(iface, ep, (struct mlx5_cqe64*)arg,
-                                        txwq, ep_status);
+        uct_dc_mlx5_iface_set_ep_failed(iface, ep, cqe, txwq, ep_status);
+    }
+
+    if (ep->dci == UCT_DC_MLX5_EP_NO_DCI) {
```

Comment:
isnt this duplication of 1324?

Reply:
you mean wrap it in oneline function?

Reply:
no; i mean that maybe we should put the code in here in the 'if' block which starts at line 1324, since it's same condition